### ISSUE TITLE
New rule: Avoid container modifications on @immutable

### ISSUE THREAD

**Issue by britov** [dart-lang/sdk#58018]
**Describe the rule you'd like to see implemented**
We have the model
```dart

@immutable
class DataState {
  final List<String> values;

  DataState(this.values);
}

```
We want to have an immutable collection inside an immutable model.

If we cannot verify the collection change inside the immutable object, we may have a problem. For example, changing currentState from Bloc (https://github.com/felangel/bloc/) (for example: https://github.com/tvip/bloc_issue_sample/blob/master/lib/main.dart)

**Examples**

https://github.com/tvip/bloc_issue_sample/blob/master/lib/main.dart

If we can’t change the collections, we must copy the collections before changing

**Comment by srawlins** [comment]
I don't think we can make a general purpose rule. Given a `Foo foo` field on an `@immutable` class, and `foo.bar()` is called, how can we know that `boo()` mutates `foo` or not?
